presto-spi/src/main/java/io/prestosql/spi/predicate/Range.java

```diff
@@ -215,4 +217,19 @@ public String toString(ConnectorSession session)
         }
         return buffer.toString();
     }
+
+    public boolean isEmpty()
+    {
+        Type type = getType();
+        Optional<Block> lowBlock = low.getValueBlock();
+        Optional<Block> highBlock = high.getValueBlock();
+        if (type.getJavaType() == long.class &&
+                low.getBound() != Marker.Bound.EXACTLY && high.getBound() != Marker.Bound.EXACTLY &&
+                low.getValueBlock().isPresent() && high.getValueBlock().isPresent()) {
+            long lowValue = type.getLong(lowBlock.get(), 0);
+            long highValue = type.getLong(highBlock.get(), 0);
+            return highValue - lowValue == 1;
```

Comment:
technically speaking, a `long`-backed type can have an ordering defined that's not compatible with ordering of `long` values used for stack representation.

it would be better to capture a relevant abstraction of a type here, eg let types define an optional method like `Optional<?> getNearestGreaterValue(Object value)`. We would implement it for intergral, decimal, date/time types. (Actually, we could implement that for real and doubles too, since effectively they are discrete types as well).

Reply:
BTW is the current code correct for REAL type?
i.e. do

- `Long::compare`
- `(long a, long b) -> `Float.compare(Float.intBitsToFloat(toIntExact(a)), Float.intBitsToFloat(toIntExact(b)))`

define the _same_ ordering?

please add relevant tests

Reply:
see also https://github.com/trinodb/trino/issues/8893